packages/elements/src/datetime-picker/index.ts

```diff
@@ -1063,7 +1063,7 @@ export class DatetimePicker extends FormFieldElement implements MultiValue {
     let dateString = '';
 
     if (inputValue) {
-      const recoveryDate = (this.interimSegments[index] || new DateTimeSegment()).getTime();
+      const recoveryDate = new Date();
```

Comment:
not using `this.interimSegments[index]` is a bad idea. Try setting value to `1450-12-12` then update from `50-12-12` to `51-12-12` with `format="yy-MM-dd"`.

This should be one of the test cases as well.